UnitTests/GitCommandsTests/Git/ExecutableExtensionsTests.cs

```diff
@@ -64,5 +67,38 @@ public void GetOutput_with_cache_miss()
             Assert.AreEqual(GitModule.SystemEncoding.GetBytes(commandOutput), outputBytes);
             Assert.IsEmpty(errorBytes);
         }
+
+        // Process argument upper bound is actually (short.MaxValue - 1)
+        // 18: '"cmd.exe" /C echo '
+        // 10: '"cmd.exe" '
+        [TestCase(short.MaxValue - 1 - 18, short.MaxValue - 1 - 18, 10)]
+        [TestCase(short.MaxValue - 1 - 19, 1, 10)]
+        public void RunBatchCommand_can_handle_max_length_arguments(int arg1Len, int arg2Len, int baseLength)
+        {
+            // Execute process
+            var exe = new Executable("cmd.exe");
+            var args = new ArgumentBuilder() { "/C echo" }
+                .BuildBatchArguments(new string[]
+                {
+                    GenerateStringByLength(arg1Len),
+                    GenerateStringByLength(arg2Len)
+                }, baseLength);
+
+            exe.RunBatchCommand(args);
```

Comment:
Please check the return value.

Reply:
Yeah the TODO above should have been left here. Will do.

Reply:
Done checking the return value, both `true` and `false`